File: Cisco/LinkedInList/Python/9MaximumNumberOfJumpsToReachTheLastIndex.py
class Solution(object):
    def maximumJumps(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: int
        """
        N = len(nums)
        dp = [-1] * N
        dp[0] = 0

        for i in range(N):  # O(N)
            if dp[i] == -1:
                continue

            for j in range(i + 1, N):  # O(N)
                if abs(nums[j] - nums[i]) <= target:
                    dp[j] = max(dp[j], dp[i] + 1)

        return dp[-1]

File: Cisco/LinkedInList/C++/9MaximumNumberOfJumpsToReachTheLastIndex.cpp
#include <vector>
#include <cmath>

using namespace std;

class Solution
{
public:
    int maximumJumps(vector<int> &nums, int target)
    {
        int N = nums.size();
        vector<int> dp(N, -1);

        dp[0] = 0;

        for (int i = 0; i < N; ++i)
        {
            if (dp[i] == -1)
                continue;

            for (int j = i + 1; j < N; ++j)
                if (target >= abs(nums[j] - nums[i]))
                    dp[j] = max(dp[j], dp[i] + 1);
        }

        return dp[N - 1];
    }
};